packages/EasyTest/src/Traits/MessengerAssertionsTrait.php

```diff
@@ -0,0 +1,211 @@
+    }
+
+    /**
+     * Asserts that the given count of messages was sent to transport.
+     *
+     * @phpstan-param class-string|null $messageClass
+     */
+    public static function assertCountOfMessagesSentToTransport(
+        int $count,
+        string $transportName,
+        ?string $messageClass = null
+    ): void {
+        Assert::assertCount($count, self::getMessagesSentToTransport($transportName, $messageClass));
+    }
+
+    /**
+     * Asserts that the given message class was dispatched by the message bus to async transport.
+     *
+     * @param mixed[] $expectedProperties
+     *
+     * @phpstan-param class-string $messageClass
+     */
+    public static function assertMessageSentToAsyncTransport(
+        string $messageClass,
+        ?array $expectedProperties = null,
+        ?int $messagesCount = null
+    ): void {
+        static::assertMessageSentToTransport($messageClass, 'async', $expectedProperties, $messagesCount);
+    }
+
+    /**
+     * Asserts that the given message class was dispatched by the message bus to failed transport.
+     *
+     * @param mixed[] $expectedProperties
+     *
+     * @phpstan-param class-string $messageClass
+     */
+    public static function assertMessageSentToFailedTransport(
+        string $messageClass,
+        ?array $expectedProperties = null,
+        ?int $messagesCount = null
+    ): void {
+        static::assertMessageSentToTransport($messageClass, 'failed', $expectedProperties, $messagesCount);
+    }
+
+    /**
+     * Asserts that the given message class was dispatched by the message bus.
+     *
+     * @param mixed[] $expectedProperties
+     *
+     * @phpstan-param class-string $messageClass
+     */
+    public static function assertMessageSentToTransport(
+        string $messageClass,
+        string $transportName,
+        ?array $expectedProperties = null,
+        ?int $messagesCount = null
+    ): void {
+        $propertyAccessor = PropertyAccess::createPropertyAccessor();
+
+        $envelopes = \array_filter(
+            self::getMessagesSentToTransport($transportName),
+            static function (object $message) use ($messageClass, $expectedProperties, $propertyAccessor) {
+                if ($message instanceof $messageClass === false) {
+                    return false;
+                }
+
+                foreach ($expectedProperties ?? [] as $property => $expectedValue) {
+                    $actualValue = $propertyAccessor->getValue($message, $property);
+
+                    if ($actualValue !== $expectedValue) {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+        );
+
+        Assert::assertCount($messagesCount ?? 1, $envelopes);
+    }
+
+    public static function consumeAsyncMessages(): void
+    {
+        self::runMessengerWorker('async', 'default');
```

Comment:
Let's extract all the magic strings into constants.

Reply:
For this, we need to add an interface or some abstract class for tests.
I think it will be overcomplicated for this case.

WDYT?

Reply:
No need for an interface, just add constants to this trait. Simply to not repeat these values through the trait code.

Reply:
I meant that this is a trait.

We can add static properties, but this also does not look good.

Reply:
Indeed, I forgot about the inability to add constants to a trait. Ok, let's keep it as is.